src/d3d9/d3d9_device.cpp

```diff
@@ -4102,12 +4102,15 @@ namespace dxvk {
 
     const Rc<DxvkBuffer> mappedBuffer = pResource->GetBuffer(Subresource);
 
-    auto formatInfo = imageFormatInfo(pResource->GetFormatMapping().FormatColor);
+    DxvkFormatInfo formatInfo = *imageFormatInfo(pResource->GetFormatMapping().FormatColor);
+    if (formatInfo.elementSize == 0)
+      formatInfo = m_adapter->GetUnsupportedFormatInfo(pResource->Desc()->Format);
```

Comment:
D3D9Device has a function to get an unsupported format. `UnsupportedFormat(pResource->Desc()->Format);`